pymeasure/instruments/redpitaya/redpitaya_scpi.py

```diff
@@ -0,0 +1,329 @@
+
+    state = Channel.control(
+        "DIG:PIN? LED{ch}", "DIG:PIN LED{ch},%d",
+        """ Control the state of the led (bool)""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, False: 0},
+    )
+
+
+class AnalogInputSlowChannel(Channel):
+    """ A slow analog input channel"""
+
+    voltage = Channel.measurement(
+        "ANALOG:PIN? AIN{ch}",
+        """ Measure the voltage on the corresponding analog input channel, range is [0, 3.3]V""",
+    )
+
+
+class AnalogOutputSlowChannel(Channel):
+    """ A slow analog output channel"""
+
+    voltage = Channel.setting(
+        "ANALOG:PIN AOUT{ch}, %f",
+        """ Set the voltage on the corresponding analog input channel, range is [0, 1.8]V""",
+        validator=truncated_range,
+        values=[0, 1.8],
+    )
+
+
+class AnalogInputFastChannel(Channel):
+
+    gain = Instrument.control(
+        "ACQ:SOUR{ch}:GAIN?", "ACQ:SOUR{ch}:GAIN %s",
+        """Control the gain of the selected fast analog input either 'LV' or 'HV' (see jumpers on boards)
+
+        'LV' set the returned values in the range [-1, 1]V and 'HV' in the range [-20, 20]V
+        """,
+        validator=strict_discrete_set,
+        values=['LV', 'HV'],
+    )
+
+    def get_data_from_ascii(self) -> np.ndarray:
+        self.write(f"ACQ:SOUR{self.id}:DATA?")
+        data_str = self.read()
+        return np.fromstring(data_str.strip('{}').encode(), sep=',')
+
+    def get_data_from_binary(self) -> np.ndarray:
+        """ Get the data in """
+        self.write(f"ACQ:SOUR{self.id}:DATA?")
+        self.parent.adapter.read_bytes(1)
+        nint = self.parent.adapter.read_bytes(1)
+        length = int(self.parent.adapter.read_bytes(nint).decode())
+        data = np.frombuffer(self.parent.adapter.read_bytes(length), dtype=int)
+        self.parent.adapter.read_bytes(2)
+        if self.gain == 'LV':
+            max_range = 2 * RedPitayaScpi.LV_MAX
+        else:
+            max_range = 2 * RedPitayaScpi.HV_MAX
+
+        return max_range * data / (2**16 - 1) - max_range / 2
+
+
+class RedPitayaScpi(Instrument):
+    """This is the class for the Redpitaya reconfigurable board
+
+    The instrument is accessed using a TCP/IP Socket communication, that is an adapter in the form:
+    "TCPIP::x.y.z.k::port::SOCKET" where x.y.z.k is the IP address of the SCPI server
+    (that should be activated on the board) and port is the TCP/IP port number, usually 5000
+
+    :parameter
+
+    """
+
+    TRIGGER_SOURCES = ['DISABLED', 'NOW', 'CH1_PE', 'CH1_NE', 'CH2_PE', 'CH2_NE',
+                       'EXT_PE', 'EXT_NE', 'AWG_PE', 'AWG_NE']
+    LV_MAX = 1
+    HV_MAX = 20
+    CLOCK = 125e6  # Hz
+
+    def __init__(self, ip_address: str, port: int = 5000, name="Redpitaya SCPI",
+                 read_termination='\r\n',
+                 write_termination='\r\n',
+                 **kwargs):
+
+        adapter = f"TCPIP::{ip_address}::{port}::SOCKET"
```

Comment:
That makes it more difficult to use other adapters (home-made or the Protocol Adapter for tests).

Best way is to give an example how to create the resource name.
You could make the `adapter` keyword optional (defaulting to None) and keep the ip_address and port.
```suggestion
    def __init__(self,
                 adapter=None,
                 ip_address: str, port: int = 5000, name="Redpitaya SCPI",
                 read_termination='\r\n',
                 write_termination='\r\n',
                 **kwargs):

        if adapter is None:
            adapter = f"TCPIP::{ip_address}::{port}::SOCKET"
```

Reply:
done as suggested